opencog/guile/SchemeSmobNew.cc

```diff
@@ -148,6 +148,95 @@ SCM SchemeSmob::handle_to_scm (const Handle& h)
 	return protom_to_scm(AtomCast(h));
 }
 
+SCM SchemeSmob::handleseq_to_scm (const HandleSeq& hs)
+{
+    SCM res = SCM_UNDEFINED;
+    bool first = true;
+    for (Handle h : hs)
+    {
+        SCM sh = handle_to_scm(h);
+        if (first) {
+            first = false;
+            res = scm_list_1(sh);
+        }
+        else {
+            res = scm_cons(sh,res);
+        }
+    }
+    return res;
+}
+
+SCM SchemeSmob::float_list_to_scm(const std::vector<double>& v)
+{
+    SCM res = SCM_UNDEFINED;
+    bool first = true;
+    for (double d : v)
+    {
+        SCM sh = scm_from_double(d);
+        if (first) {
+            first = false;
+            res = scm_list_1(sh);
+        }
+        else {
+            res = scm_cons(sh,res);
+        }
+    }
+    return res;
+}
+
+SCM SchemeSmob::dvkey_to_scm(const DVKey& v)
+{
+    SCM res = SCM_UNDEFINED;
+    bool first = true;
+    for (auto d : v)
+    {
+        SCM sh = float_list_to_scm(d);
+        if (first) {
+            first = false;
+            res = scm_list_1(sh);
+        }
+        else {
+            res = scm_cons(sh,res);
+        }
+    }
+    return res;
+}
+
+SCM SchemeSmob::dvkeyseq_to_scm(const DVKeySeq& v)
+{
+    SCM res = SCM_UNDEFINED;
+    bool first = true;
+    for (auto d : v)
+    {
+        SCM sh = dvkey_to_scm(d);
+        if (first) {
+            first = false;
+            res = scm_list_1(sh);
+        }
+        else {
+            res = scm_cons(sh,res);
+        }
+    }
+    return res;
+}
+
+SCM SchemeSmob::protomseq_to_scm (const ProtomSeq& ps)
```

Comment:
why is this, and all the above needed?  Its adding a bunch of code that .. It seems to be useless, un-needed code. 

Maintaining code that no one uses, no one needs creates a burden for future maintainers: raises the bar for learning what the hell it does,  and general mental overload for remembering what is where, why, how.  Increased mental burden non developers means increased bugs, glitches, security crashes, etc.